pkg/kubecost/allocation_json.go

```diff
@@ -1,59 +1,130 @@
+	Minutes                    *float64               `json:"minutes"`
+	CPUCores                   *float64               `json:"cpuCores"`
+	CPUCoreRequestAverage      *float64               `json:"cpuCoreRequestAverage"`
+	CPUCoreUsageAverage        *float64               `json:"cpuCoreUsageAverage"`
+	CPUCoreHours               *float64               `json:"cpuCoreHours"`
+	CPUCost                    *float64               `json:"cpuCost"`
+	CPUCostAdjustment          *float64               `json:"cpuCostAdjustment"`
+	CPUEfficiency              *float64               `json:"cpuEfficiency"`
+	GPUCount                   *float64               `json:"gpuCount"`
+	GPUHours                   *float64               `json:"gpuHours"`
+	GPUCost                    *float64               `json:"gpuCost"`
+	GPUCostAdjustment          *float64               `json:"gpuCostAdjustment"`
+	NetworkTransferBytes       *float64               `json:"networkTransferBytes"`
+	NetworkReceiveBytes        *float64               `json:"networkReceiveBytes"`
+	NetworkCost                *float64               `json:"networkCost"`
+	NetworkCrossZoneCost       *float64               `json:"networkCrossZoneCost"`
+	NetworkCrossRegionCost     *float64               `json:"networkCrossRegionCost"`
+	NetworkInternetCost        *float64               `json:"networkInternetCost"`
+	NetworkCostAdjustment      *float64               `json:"networkCostAdjustment"`
+	LoadBalancerCost           *float64               `json:"loadBalancerCost"`
+	LoadBalancerCostAdjustment *float64               `json:"loadBalancerCostAdjustment"`
+	PVBytes                    *float64               `json:"pvBytes"`
+	PVByteHours                *float64               `json:"pvByteHours"`
+	PVCost                     *float64               `json:"pvCost"`
+	PVs                        PVAllocations          `json:"pvs"`
+	PVCostAdjustment           *float64               `json:"pvCostAdjustment"`
+	RAMBytes                   *float64               `json:"ramBytes"`
+	RAMByteRequestAverage      *float64               `json:"ramByteRequestAverage"`
+	RAMByteUsageAverage        *float64               `json:"ramByteUsageAverage"`
+	RAMByteHours               *float64               `json:"ramByteHours"`
+	RAMCost                    *float64               `json:"ramCost"`
+	RAMCostAdjustment          *float64               `json:"ramCostAdjustment"`
+	RAMEfficiency              *float64               `json:"ramEfficiency"`
+	ExternalCost               *float64               `json:"externalCost"`
+	SharedCost                 *float64               `json:"sharedCost"`
+	TotalCost                  *float64               `json:"totalCost"`
+	TotalEfficiency            *float64               `json:"totalEfficiency"`
+	RawAllocationOnly          *RawAllocationOnlyData `json:"rawAllocationOnly"`
+}
+
+func (aj *AllocationJSON) BuildFromAllocation(a *Allocation) {
+	if aj == nil {
+		return
+	}
+	aj.Name = a.Name
```

Comment:
Is it fair to say that the original bug was entirely a function of our bad JSON func? So we don't actually have to take any extra action here?

Reply:
Correct. With this code I can no longer reproduce the original issue.